.github/workflows/ci.yml

```diff
@@ -2,145 +2,186 @@ name: Continuous Integration
 
 on: [push, pull_request, workflow_dispatch]
 
+env:
+  TEST_ALL_DOCKER_IMAGE: 'parsec-service-test-all'
+  # TEST_ALL_DOCKER_IMAGE: 'ghcr.io/parallaxsecond/parsec-service-test-all'
+
 jobs:
+  build-and-export-test-all-docker:
+    runs-on: ubuntu-latest
+    # If TEST_ALL_DOCKER_IMAGE is 'parsec-service-test-all' or any local image,
+    # the following condition must evaluate true to execute this job
+    # Else it must evaluate false to NOT execute this job
+    # Unfortunately, env.TEST_ALL_DOCKER_IMAGE cannot be used here as the `env` context is not recognize at this level.
+    if: ${{ true }} # env.TEST_ALL_DOCKER_IMAGE == 'parsec-service-test-all'
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build the docker container
+      run: pushd e2e_tests/docker_image && docker build -t parsec-service-test-all -f parsec-service-test-all.Dockerfile . && popd
+    - name: Export the docker container
+      run: docker save parsec-service-test-all > /tmp/parsec-service-test-all.tar
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: parsec-service-test-all
+        path: /tmp/parsec-service-test-all.tar
```

Comment:
So what's the plan with this job? It seems to be useful when some change is made to the Docker image, but less than ideal if it runs all the time.

Reply:
Yes, that's why there is a condition on the execution. Unfortunately, I couldn't use the environment variable in this context.
So whenever we have docker changes, we switch only two lines
- the name of the image `parsec-service-test-all`
- this job condition to `true`
default will be `ghcr.io/parallaxsecond/parsec-service-test-all`
and condition will be `false`

Reply:
Cool, thanks, that's what I was hoping for!